### DISCUSSION TITLE
ScriptReference/Pool.ListPool_1

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#91] [Notes]
# ScriptReference/Pool.ListPool_1

https://docs.unity3d.com/ScriptReference/Pool.ListPool_1.html

**Comment by vertxxyz** [comment]
Note that because anything can `Get` a List and these lists are never removed from memory (they are static), if a section of code is growing list capacity it can affect all Lists in the pool. As the list capacity grows it can walk across memory and cause fragmentation.
If you're facing memory issues on a platform with tight requirements (eg. Nintendo Switch), use the [Memory Profiler](https://docs.unity3d.com/Packages/com.unity.memoryprofiler@latest/) to confirm that fragmentation is your issue, hunt down the list pools and create a number of suitably max-capacity pooled lists at startup.
